### PR TITLE
Integrated Wikimedia OAuth

### PR DESCRIPTION
**Pull request by Parthiv-M**
Integrated Wikimedia OAuth functionality with Codex components

Bug: [T337927](https://phabricator.wikimedia.org/T337927)

### CODE REVIEW

**Comment by coderwassananmol** [comment]
@Parthiv-M I've pushed the design changes to the branch. Please pull the changes and re-submit the PR.

File: hooks/withSession.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useSession } from "next-auth/react";
+
+export const withSession = (Component) => (props) => {
+  const session = useSession();
+
+  if (Component.prototype.render) {
+    return <Component session={session} {...props} />;
+  }
+
+  throw new Error("Use `useSession` directly");
```
Comment (coderwassananmol) [inline hooks/withSession.js:10]: Log the session errors in winston.

File: package.json [hunk 2]
```diff
@@ -45,9 +46,13 @@
     "jsdom": "^20.0.0",
     "jspdf": "^2.5.1",
     "jszip": "^3.2.2",
+    "less": "^4.1.3",
+    "less-loader": "^11.1.3",
     "loaders.css": "^0.1.2",
     "lodash": "^4.17.20",
     "next": "^12.2.5",
+    "next-auth": "^4.22.3",
```
Comment (coderwassananmol) [inline package.json:54]: As per https://stackoverflow.com/a/74340164, the current version of `next-auth` won't work with `next^12.2.5`. Please ensure the dependencies are met.

**Review by coderwassananmol — CHANGES_REQUESTED** [review]

File: hooks/withSession.js (hunk 1 above)
Comment (Parthiv-M) [reply hooks/withSession.js]: Winston [cannot log in the front end](https://github.com/winstonjs/winston-daily-rotate-file/issues/339#issuecomment-1182901700), so I have refrained from doing it as of now. Moreover, `Error("Use useSession directly")` is not about the session itself, hence I decided not to include it.

**Review by coderwassananmol — APPROVED** [review]

**Pull request merged**